Style Guide fixes and tweaks

benknoble opened the pull request:
##### Checklist

- [x] Bugfix
- ~[ ] Feature~
- ~[ ] tests included~
- [x] documentation

### Description of change

This PR proposes a set of independent tweaks to the [Style Guide](https://docs.racket-lang.org/style/index.html).
Feel free to cherry-pick the ones you like and ignore those you don't.

sorawee commented on pkgs/racket-doc/scribblings/style/textual.scrbl:
```diff
@@ -144,10 +144,11 @@ read code on monitors that accommodate close to 250 columns, and on
-These lines help both writers and readers to orient themselves in a file.
-In scribble use @litchar|{@; }| as the prefix.
+So, when you create a file, add a line with @litchar{;; } followed by ctrl-U 99
+and @litchar{-}. @margin-note{In Vi, the command is 99a- followed by Esc.} When
```
How does this render? I recall there's a difference between `margin-note` and `margin-note*`. Is this the right one to use?

benknoble replied:
Ah, I think you're right that this should be `margin-note*` given its use mid-paragraph. Will fix and check rendering.

benknoble replied:
You were correct; fixed. Range-diff (not hidden because it's small):
```
1:  907d3ea366 = 1:  907d3ea366 docs: capitalize the noun Git
2:  1d248982d7 ! 2:  665e037505 docs: mention the Vi command to add sections
    @@ pkgs/racket-doc/scribblings/style/textual.scrbl: read code on monitors that acco
     -These lines help both writers and readers to orient themselves in a file.
     -In scribble use @litchar|{@; }| as the prefix.
     +So, when you create a file, add a line with @litchar{;; } followed by ctrl-U 99
    -+and @litchar{-}. @margin-note{In Vi, the command is 99a- followed by Esc.} When
    ++and @litchar{-}. @margin-note*{In Vi, the command is 99a- followed by Esc.} When
     +you separate "sections" of code in a file, insert the same line. These lines
     +help both writers and readers to orient themselves in a file. In scribble use
     +@litchar|{@; }| as the prefix.
3:  adce70f37d = 3:  808676897e docs: link fx+ and unsafe-fx+
4:  7823976663 = 4:  ca7d2a2a56 docs: correct Git pull command
5:  7521dcb753 = 5:  1108c95343 docs: unquote "merge commit"
6:  af33479301 = 6:  1374b3b095 docs: italicize "e.g."
7:  b082353e9d = 7:  8f3f1cd517 docs: correct macro body
```

mflatt commented:
Thanks for these improvements!

But can we skip the "italicize e.g." change and instead update section 5.1 to say that common Latin phrases/abbreviations should *not* be italicized? Although I saw an italicization go by recently and didn't bother to object, there must be many unitalicized instances in the documentation. So there's a convention pretty well in place, and I'm skeptical that italicizing makes things clearer so that we should change.

rfindler commented:
> But can we skip the "italicize e.g." change and instead update section 5.1 to say that common Latin phrases/abbreviations should _not_ be italicized? 

+1 from me on this one. I never understood the phrases-in-foreign-languages-should-be-italicized rule. Why ``ad hoc'' and not ``cuisine''? Where do you draw the line when most of the language is taken from other languages? Not to mention that random italics sprinkled though a text looks bad and you lose the ability to use italics to draw the eye (which is what they do, after all)  okay, \<\/rant\>

mfelleisen commented:
I use italics for latin phrases only, because style guides in the 1980s suggested this. 

Other foreign words I don’t use. (I don’t even think of “cuisine” as a foreign word, 
but that’s perhaps due to my personal background.)

rfindler commented:
> Other foreign words I don’t use. (I don’t even think of “cuisine” as a foreign word, but that’s perhaps due to my personal background.)

Yeah, I was trying to make the point that, although “cuisine” doesn't feel like a foreign word it also feels very French, at least to me. Given “ad hoc”, tho, maybe I should have used “atrium” instead! How about “ergo” or “etcetera”? 😄

benknoble commented:
> But can we skip the "italicize e.g." change and instead update section 5.1 to say that common Latin phrases/abbreviations should _not_ be italicized? Although I saw an italicization go by recently and didn't bother to object, there must be many unitalicized instances in the documentation. So there's a convention pretty well in place, and I'm skeptical that italicizing makes things clearer so that we should change.

Done, although I've put it in 5.3 (Typesetting Prose), where it seemed like a better fit?

<details><summary>range-diff: rebased on latest; dropped italics; added note about Latin phrases & abbreviations</summary>

```
1:  907d3ea366 = 1:  945d895934 docs: capitalize the noun Git
2:  665e037505 = 2:  d9943998ca docs: mention the Vi command to add sections
3:  808676897e = 3:  01b511b849 docs: link fx+ and unsafe-fx+
4:  ca7d2a2a56 = 4:  f12b1170c5 docs: correct Git pull command
5:  1108c95343 = 5:  78968271a5 docs: unquote "merge commit"
6:  1374b3b095 < -:  ---------- docs: italicize "e.g."
7:  8f3f1cd517 = 6:  8cf1dc6855 docs: correct macro body
-:  ---------- > 7:  090fef892e docs: make explicit the convention for Latin
```

</details>

mfelleisen commented:
Thanks for paying close attention and fixing these flaws.

sorawee commented on pkgs/racket-doc/scribblings/style/textual.scrbl:
```diff
@@ -144,10 +144,11 @@ read code on monitors that accommodate close to 250 columns, and on
+So, when you create a file, add a line with @litchar{;; } followed by ctrl-U 99
+and @litchar{-}. @margin-note*{In Vi, the command is 99a- followed by Esc.} When
+you separate "sections" of code in a file, insert the same line. These lines
+help both writers and readers to orient themselves in a file. In scribble use
```
This is probably out of scope for the PR (so feel free to ignore this comment), but "scribble" probably should be capitalized (there might be other occurrences too?).

benknoble replied:
I should at least fix new occurrences ;)

benknoble replied:
Fixed.

<details><summary>range-diff</summary>

```
1:  945d895934 = 1:  35c19f1e83 docs: capitalize the noun Git
2:  d9943998ca ! 2:  6088cd1567 docs: mention the Vi command to add sections
    @@ pkgs/racket-doc/scribblings/style/textual.scrbl: read code on monitors that acco
     +So, when you create a file, add a line with @litchar{;; } followed by ctrl-U 99
     +and @litchar{-}. @margin-note*{In Vi, the command is 99a- followed by Esc.} When
     +you separate "sections" of code in a file, insert the same line. These lines
    -+help both writers and readers to orient themselves in a file. In scribble use
    ++help both writers and readers to orient themselves in a file. In Scribble use
     +@litchar|{@; }| as the prefix.
      
      @; -----------------------------------------------------------------------------
3:  01b511b849 = 3:  1e7b35da0a docs: link fx+ and unsafe-fx+
4:  f12b1170c5 = 4:  c3e32a5afa docs: correct Git pull command
5:  78968271a5 = 5:  372bbd4ad5 docs: unquote "merge commit"
6:  8cf1dc6855 = 6:  e48525eeb7 docs: correct macro body
7:  090fef892e = 7:  38b3c0a75e docs: make explicit the convention for Latin
```

</details>
